### ISSUE TITLE
Migration Details Page Should Auto-Refresh

### ISSUE THREAD

**Issue by kedark3** [ManageIQ/miq_v2v_ui_plugin#311]
https://bugzilla.redhat.com/show_bug.cgi?id=1573625
Since we have a spinner on the page which gives false impression of page will be automatically refreshed, but it does not.

**Comment by kedark3** [comment]
@priley86 @AparnaKarve I am happy to collaborate. Let me know what you need. :smiley:

**Comment by mturley** [comment]
Grabbing this as soon as #322 is merged.

**Comment by serenamarie125** [comment]
Here's the image attached to the gif

![peek 2018-05-17 17-08](https://user-images.githubusercontent.com/12504403/40211787-9ca4826a-5a1a-11e8-8b0f-16ef4211f36b.gif)

**Comment by serenamarie125** [comment]
@vconzola @priley86 it is a Failed Migration Plan that is being looked at.  I asked for some clarification.

No auto refresh is necessary because this is a failed migration plan, and the migration of the VM has already failed.  

Looking at the details page, there is a utilization bar, which behaves like a progress bar.  I'm wondering if this is the issue, from the reporters perspective.  It does look like it's still in progress.

I've asked for more information, so let' see what he has to say.

**Comment by kedark3** [comment]
@Serena,
I agree that this is a failed migration. But if you navigate to same screen while **migration is in progress,** the utilization bar does not update dynamically.You need to hit refresh manually to see progress in the bar, as well as the clock for elapsed time increase. This all should update without needing to hit refresh on the page. 
Hope that helps.

**Comment by serenamarie125** [comment]
ahhh ok @kedark3 !  Thank you for the info.

@mturley please try and reproduce this when looking at the details of a migration in progress.  Sit on that screen, and please verify that the utilization bar is not dynamically updating.  If you cannot reproduce on your local environment, please use the environment Brett gave us access to.  Please add a note to this Issue once you have tried to reproduced the issue

**Comment by mturley** [comment]
Thanks for the clarification @kedark3 and for the mention @serenamarie125. I tried to reproduce once and didn't see the issue, I'll take another look on Monday morning with this in mind.

**Comment by mturley** [comment]
@kedark3 I agree that the time and progress bar should update in real-time, even though the polling is on every 15 seconds. I'm trying to reproduce the issue now, and I think we will add that real-time elapsing to the timer, but does the progress bar not update for you after waiting longer than 15 seconds? Checking on your environment right now myself as well.

**Comment by kedark3** [comment]
@mturley Yes Mike, the progress bar does not update. Even if I wait.

**Comment by JPrause** [comment]
This is a blocker for issue: https://bugzilla.redhat.com/show_bug.cgi?id=1573625

**Comment by mturley** [comment]
@kedark3 it appears to update for me if I wait long enough. What browser are you using? do you see any errors in your JS console?

http://recordit.co/Mr9HlkXe7a

**Comment by kedark3** [comment]
@mturley you should see Migration Details Page  which is https://10.8.196.145/migration/plan/6, it does not update

**Comment by mturley** [comment]
Sorry for confusion, yes, that page appears to have a bug. I have a potential fix, I'm trying to figure out how best to test it. We may just ship it and see-- i think it's a harmless change, @priley86 can you look at this for me? https://github.com/ManageIQ/miq_v2v_ui_plugin/pull/339

I think if that PR doesn't fix it, we have a remounting issue or some other more serious race condition.

**Comment by mturley** [comment]
@AparnaKarve mentioned she is making some time for this issue later today. Aparna maybe you can pick up where i left off with https://github.com/ManageIQ/miq_v2v_ui_plugin/pull/339

**Comment by kedark3** [comment]
@mturley If you want me to test it, including it in nightly build will be most reliable and easy way for testing it.

**Comment by mturley** [comment]
@priley86 I think your closing of my PR auto-closed this because I mentioned it with the magic word "fix". Reopening until @kedark3 has a chance to test it. This will be in the nightly, @kedark3, let us know if it solves the problem for you!

**Comment by kedark3** [comment]
@mturley I will update you on this when nightly becomes available. Nightly from yesterday is not available as it seems.

**Comment by mturley** [comment]
Hmm ok. Thanks @kedark3. We are fairly confident this nightly will fix the issue, but you are our fastest way to test it right now! Thanks for your help.

**Comment by priley86** [comment]
will wait on @bthurber to confirm this is available for QA  - should be g2g for nightly
https://github.com/ManageIQ/manageiq/pull/17442

**Comment by bthurber** [comment]
Deployed last evening's nightly and not seeing dynamic updates in the migration plan details. Appears to still be broken.

**Comment by priley86** [comment]
Ok thanks @bthurber. Will continue investigating if we can login to the MIQ instance there. Apologies...

**Comment by JPrause** [comment]
@miq-bot add_label blocker

**Comment by mturley** [comment]
That's disconcerting. @priley86 I could use help debugging this one if it's not solved by that fix. Can we confirm that the PR made it into the nightly they used?

**Comment by AparnaKarve** [comment]
@mturley I will be trying the following change on Brett's machine shortly...

```js
this.startPolling(planRequestId);
if (mostRecentRequest.fulfilled_on !== null) {
  this.stopPolling();
  this.setState(() => ({
  planFinished: true
 }));
}
```

**Comment by AparnaKarve** [comment]
@mturley @priley86 News flash!!
Just looked into the code on Brett's machine and it *still* shows the old code -
```js
if (mostRecentRequest.status === 'active') {
          this.startPolling(planRequestId);
        } else if (
          mostRecentRequest.request_state === 'finished' ||
          mostRecentRequest.status === 'Error' 
        ) {     
          this.setState(() => ({
            planFinished: true
          }));    
        } 
```

It looks like our changes in this area never made it to the nightly build.

However, I do think that the above code change https://github.com/ManageIQ/miq_v2v_ui_plugin/issues/311#issuecomment-391791426 is necessary.

**Comment by mturley** [comment]
Thanks @AparnaKarve . If you need to work on anything else let me know and I'll take over again. I want to make sure we get this fixed asap.

**Comment by mturley** [comment]
Hmm.... I wonder why that is.... I thought I had updated everything. I'll take another look.

**Comment by mturley** [comment]
@AparnaKarve I think the code change we did check in will fix the issue, but I agree yours seems more reliable. Are you making another PR?  I'll look into why my old gemfile is still in there.

**Comment by mturley** [comment]
Ahh, that makes more sense. I thought you had seen that in my PR or I had otherwise missed s something.

Thanks again @AparnaKarve let me know if there is anything I can do to help.

**Comment by AparnaKarve** [comment]
@mturley I have confirmed on the appliance that the page refreshes correctly with the latest master.

I don't think we need an additional PR to do this - https://github.com/ManageIQ/miq_v2v_ui_plugin/issues/311#issuecomment-391791426

There is one thing missing though on the Plan detail page --  the notification popup when the migration completes, for which I can make a PR -- should be fairly simple I think.

**Comment by kedark3** [comment]
@AparnaKarve is it already in nightly? if so I can try to validate it.

**Comment by AparnaKarve** [comment]
@kedark3 I don't know if the nightly build issue is fixed or not.

On your appliance, can you check the "About" box and see what MIQ version and date it shows?

**Comment by kedark3** [comment]
Its still old. :(

**Comment by JPrause** [comment]
@AllenBW can you add the blocker label for me,...there's a nickel in it for you.

**Comment by bthurber** [comment]
Verified this is fixed in latest nightly.

**Comment by serenamarie125** [comment]
Fixed by #339 🎉🍻

**Comment by bthurber** [comment]
@mturley confirmed this is not fixed in current 5.9.3.1 downstream build.  Need to verify commit is pushed there so we can unblock the next 5.9.3.1 build.

**Comment by mturley** [comment]
Hmm. Ok. It was fixed in the nightly, though? Not sure how to check the downstream build, how can I help?
